scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public Serializable compute(String originator, TaskId taskid, String channel, String key, String remappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
+        Serializable answer;
+        try {
+            answer = getChannel(channel).compute(key,
+                                                 (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(remappingFunction,
+                                                                                                                                            BiFunction.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run compute" + ON_KEY + QUOTE + key + QUOTE + RESULT_IS + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+
+        return answer;
+    }
+
+    @Override
+    public Serializable merge(String originator, TaskId taskid, String channel, String key, Serializable value,
+            String remappingFunction) throws InvalidChannelException, CompilationException, IOException {
+        Serializable answer;
+        try {
+            answer = getChannel(channel).merge(key,
+                                               value,
+                                               (BiFunction<? super Serializable, ? super Serializable, ? extends Serializable>) evaluateClosure(remappingFunction,
+                                                                                                                                                BiFunction.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run merge" + ON_KEY + QUOTE + key + QUOTE + WITH_VALUE + value + RESULT_IS + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public PredicateActionResult conditionalCompute(String originator, TaskId taskid, String channel, String key,
+            String predicate, String thenRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)